storage/diff_scheme/diff_types.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <unordered_map>
+
+namespace diff_scheme
+{
+enum class Status
+{
+  Undefined,
+  NotAvailable,
+  Available
+};
+
+struct FileInfo final
+{
+  FileInfo(uint64_t size, uint64_t version) : m_size(size), m_version(version) {}
+  uint64_t m_size;
```

Comment:
I can't see any reasons to set it because you can't create an instance of FileInfo in any way except using the FileInfo ctor with two parameters.